src/safe_apps/tests/test_views.py

```diff
@@ -690,3 +771,39 @@ def test_query_url_no_match(self) -> None:
         json_response = response.json()
         self.assertEqual(response.status_code, 200)
         self.assertTrue(len(json_response) == 0)
+
+
+class SafeAppsSocialProfilesTests(APITestCase):
+    def test_empty_social_profiles(self) -> None:
+        SafeAppFactory.create()
+        url = reverse("v1:safe-apps:list")
+
+        response = self.client.get(path=url, data=None, format="json")
+
+        json_response = response.json()
+        self.assertEqual(response.status_code, 200)
+        self.assertEqual(json_response[0]["socialProfiles"], [])
+
+    def test_multiple_social_profiles(self) -> None:
+        safe_app = SafeAppFactory.create()
+        social_profile_1 = SocialProfileFactory.create(platform="Z", safe_app=safe_app)
+        social_profile_2 = SocialProfileFactory.create(platform="A", safe_app=safe_app)
```

Comment:
Since the factories don't perform a `full_clean` the `SocialProfile` will be successfully created even though the platforms are not part of our `TextChoices`.

We should test that by using the respective social networks that we support that these are currently rendered on the client side. That includes things like:
- If the text choice is `DISCORD` that `DISCORD` is correctly serialized as a string in the payload (and the same of other networks)
- How do we render Safe Apps with multiple social profiles of the same type.
- How do we render Safe Apps with multiple social profiles of different types.


However, `TextChoices` are not validated by default so that's why this test works right now. I suggest adding something to the Factory like:

```python
@factory.post_generation
def validate(obj, create, extracted, **kwargs):
    if not create:
        return
    obj.full_clean()
```

By doing this, this test should fail.

Reply:
Thanks! Good catches, have just addressed them in my new commit